src/mmrelay/meshtastic_utils.py

```
@@ -250,6 +251,57 @@ def _get_metadata_executor() -> ThreadPoolExecutor:
+
+    This uses the same `get_device_metadata_request` admin packet as
+    `localNode.getMetadata()`, but avoids the library's print-based response
+    handler so health checks do not redirect process-wide stdout.
+    """
+    local_node = getattr(client, "localNode", None)
+    if (
+        local_node is None
+        or not hasattr(local_node, "_sendAdmin")
+        or not hasattr(client, "waitForAckNak")
+    ):
+        raise RuntimeError("Meshtastic client cannot perform metadata liveness probe")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)